apis/analysis/v1alpha1/recommendation_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RecommendationTargetType defines the type of analysis target
+type RecommendationTargetType string
+
+const (
+	// RecommendationTargetWorkload defines the k8s workload type
+	RecommendationTargetWorkload RecommendationTargetType = "workload"
+	// RecommendationPodSelector defines the pod selector type
+	RecommendationPodSelector RecommendationTargetType = "podSelector"
+)
+
+// RecommendationTarget defines the target of analysis, which can be a k8s workload or a series of pods
+type RecommendationTarget struct {
+	// Type indicates the type of target
+	Type RecommendationTargetType `json:"type"`
+	// Workload indicates the target is a k8s workload, which is effective when Type is "workload"
+	Workload *CrossVersionObjectReference `json:"workload,omitempty"`
+	// PodSelector defines the reference of a series of pods, which is effective when Type is "podSelector"
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+
+// CrossVersionObjectReference contains enough information to let you identify the referred resource.
+type CrossVersionObjectReference struct {
```

Comment:
Shall we reuse the `autoscalingv2.CrossVersionObjectReference`?

Reply:
we notice the autoscalingv2 but do not want to import unnecessary dependencies, especially this may be extended in future.